snap-gpf-ui/src/main/java/org/esa/snap/core/gpf/ui/resample/ResamplingDialog.java

```diff
@@ -417,9 +417,9 @@ private void updateTargetResolutionTargetWidthAndHeight() {
 
         private void reactToSourceProductChange(Product product) {
             if (product != null) {
-                resolutionSpinner.setValue(determineResolutionFromProduct(product));
+                resolutionSpinner.setValue(new Double(determineResolutionFromProduct(product)));
```

Comment:
Rather than creating a Double object here, make determineResolutionFromProduct return a double in the first place.